core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -815,6 +815,39 @@ public static int bitLength(ByteString s) {
     return s.length() * 8;
   }
 
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(long value, int position) {
+    checkPosition(position, java.lang.Long.SIZE);
+    return (int) ((value >> position) & 1);
+  }
+
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(int value, int position) {
+    checkPosition(position, java.lang.Integer.SIZE);
+    return (value >> position) & 1;
+  }
+
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(short value, int position) {
+    checkPosition(position, java.lang.Short.SIZE);
+    return (value >> position) & 1;
+  }
+
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(byte value, int position) {
+    checkPosition(position, java.lang.Byte.SIZE);
+    return (value >> position) & 1;
+  }
+
+  private static void checkPosition(int position, int size) {
+    if (position < 0) {
+      throw RESOURCE.illegalNegativeBitGetPosition().ex();
+    }
+    if (size <= position) {
+      throw RESOURCE.illegalBitGetPositionExceedsLimit().ex();
```

Comment:
It would be better to add `size` and `position` in the error message

Reply:
Good advice, I will update it soon.